### PR TITLE
Uses scripted syntax for Jenkinsfile for GitHub hook compat

### PR DESCRIPTION
**Pull request by evindunn**
## Description

This Jenkinsfile uses the newer 'scripted' syntax, as opposed to the 'declarative' syntax. More info [here](https://jenkins.io/doc/book/pipeline/syntax/#compare). This newer syntax is used for compatibility with the GitHub plugin, which fails to associate webhooks with Jenkins jobs when the old syntax is used. The magic is line 1. After a single manual Jenkins build, line 1 registers the Jenkins Job with the repository that triggered the webhook. For subsequent pushes to the repository, Jenkins will receive the POST from GitHub and use it to trigger the job.

## Related Issue

Internal Jenkins-dev builds weren't being triggered by the GitHub webhook

## Motivation and Context

Automated builds/testing for isis3 on centos 7, debian 9, fedora 25, ubuntu 18.04

## How Has This Been Tested?

Tested with manual builds in jenkins-dev

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ x ] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [ x ] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [ x ] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Comment by Kelvinrr** [comment]
Once we got this tested by using the pipeline directly in Jenkins with PR and Push support, we should be good.

File: Jenkinsfile [hunk 1]
```diff
@@ -1,64 +1,93 @@
...
+
+def nodes = [:]
+
+nodes["isis-fedora-25"] = {
+    node("isis-fedora-25") {
+        withEnv(["ISISROOT=" + "${workspace}" + "/build/", "ISIS3TESTDATA=/usgs/cpkgs/isis3/testData/", "ISIS3DATA=/usgs/cpkgs/isis3/data/"]) {
+            stage ("Fedora 25") {
+                git branch: 'dev', url: 'https://github.com/USGS-Astrogeology/ISIS3.git'
+                sh """
+                    conda env create -n isis3 -f environment.yml
```
Comment (Kelvinrr) [inline Jenkinsfile:15]: We will need an additional line to pull the gtest submodule

**Comment by Kelvinrr** [comment]
Can one of the admins verify this patch?

File: Jenkinsfile (hunk 1 above)
Comment (evindunn) [reply Jenkinsfile:15]: Done

**Review by jlaura — APPROVED** [review]

**Pull request merged**